src/admin.rs

```diff
@@ -1340,3 +1378,184 @@ impl Future for AlterConfigsFuture {
+}
+
+#[derive(Debug)]
+pub struct TopicPartitionInfo {
+    pub partition: i32,
+    pub leader: Node,
+    pub isr: Vec<Node>,
+    pub replicas: Vec<Node>,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum AclOperation {
+    Unknown,
+    Any,
+    All,
+    Read,
+    Write,
+    Create,
+    Delete,
+    Alter,
+    Describe,
+    ClusterAction,
+    DescribeConfigs,
+    AlterConfigs,
+    IdempotentWrite,
+}
+
+#[derive(Debug)]
+pub struct TopicDescription {
+    pub name: String,
+    pub topic_id: Uuid,
+    pub partitions: Vec<TopicPartitionInfo>,
+    pub is_internal: bool,
+    pub authorized_operations: Vec<AclOperation>,
+}
+
+type NativeTopicCollection = NativePtr<RDKafkaTopicCollection>;
+
+unsafe impl KafkaDrop for RDKafkaTopicCollection {
+    const TYPE: &'static str = "topic collection";
+    const DROP: unsafe extern "C" fn(*mut Self) = rdsys::rd_kafka_TopicCollection_destroy;
+}
+
+
+struct DescribeTopicsFuture {
+    rx: oneshot::Receiver<NativeEvent>,
+}
+
+impl Future for DescribeTopicsFuture {
+    type Output = KafkaResult<Vec<TopicDescription>>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let event = ready!(self.rx.poll_unpin(cx)).map_err(|_| KafkaError::Canceled)?;
+        event.check_error()?;
+        let res = unsafe { rdsys::rd_kafka_event_DescribeTopics_result(event.ptr()) };
+        if res.is_null() {
+            let typ = unsafe { rdsys::rd_kafka_event_type(event.ptr()) };
+            return Poll::Ready(Err(KafkaError::AdminOpCreation(format!(
+                "describe topics request received response of incorrect type ({})",
+                typ
+            ))));
+        }
+        let mut n = 0;
+        let topic_descriptions = unsafe { rdsys::rd_kafka_DescribeTopics_result_topics(res, &mut n) };
+        let mut out = Vec::with_capacity(n);
+        for i in 0..n {
+            let topic_description = unsafe { *topic_descriptions.add(i) };
+            let topic_description = TopicDescription {
+                name: unsafe { cstr_to_owned(rdsys::rd_kafka_TopicDescription_name(topic_description)) },
+                topic_id: extract_topic_id(topic_description),
+                partitions: extract_partitions(topic_description),
+                is_internal: unsafe { rdsys::rd_kafka_TopicDescription_is_internal(topic_description) } != 0,
+                authorized_operations: extract_authorized_operations(topic_description)?,
+            };
+            out.push(topic_description);
+        }
+        Poll::Ready(Ok(out))
+    }
+}
+
+fn extract_topic_id(topic_description: *const RDKafkaTopicDescription) -> Uuid {
+    let topic_id = unsafe { rdsys::rd_kafka_TopicDescription_topic_id(topic_description) };
+    let high_bits = unsafe { rdsys::rd_kafka_Uuid_most_significant_bits(topic_id) } as u64;
+    let low_bits = unsafe { rdsys::rd_kafka_Uuid_least_significant_bits(topic_id) } as u64;
+    Uuid::from_u64_pair(high_bits, low_bits)
+}
```

Comment:
Is it really worth adding a dependency to the `uuid` crate when `Uuid` is only used here? We might as well remove the dependency and only return a `(high_bits, low_bits)` tuple.